library/src-bootstrapped/scala/StagedTuple.scala

```diff
@@ -122,11 +122,11 @@ object StagedTuple {
   }
 
   def applyStaged[Tup <: NonEmptyTuple : Type, N <: Int : Type](tup: Expr[Tup], size: Option[Int], n: Expr[N], nValue: Option[Int]): Expr[Elem[Tup, N]] = {
-    if (!specialize) '{dynamicApply($tup, $n)}
+    if (!specialize) '{dynamicApply($tup, $n)}.asInstanceOf // TODO: check that variance is OK here
     else {
       def fallbackApply(): Expr[Elem[Tup, N]] = nValue match {
         case Some(n) => quoted.QuoteError("index out of bounds: " + n)
-        case None => '{dynamicApply($tup, $n)}
+        case None => '{dynamicApply($tup, $n)}.asInstanceOf // TODO: check that variance is OK here
```

Comment:
I think you might be able to avoid the casts by changing the signature of `dynamicApply` to the following:

```scala
def dynamicApply[This <: NonEmptyTuple, N <: Int] (self: This, n: N): Elem[This, N] = {
  type Result = Elem[This, N]
  ...
}
```

Reply:
That works 👍